homeassistant/config_entries.py

```diff
@@ -706,6 +712,18 @@ async def async_setup(
                 error_reason_translation_placeholders,
             )
 
+    @callback
+    def _raise_for_exception_in_forwarded_platform(
+        self, exc: Exception, integration: loader.Integration
+    ) -> None:
+        """Raise exception if the wrong exception is raised in a forwarded platform."""
+        report(
+            f"raises exception {type(exc)} {exc} in forwarded platform"
```

Comment:
Should we clarify right behavior instead?

```suggestion
            f"raises exception {type(exc)} {exc} in forwarded platform. Raise before forwarding instead."
```

Reply:
Yes absolutely.  Will cleanup the message a bit more when I add the tests.

I only found this by happenstance when I was looking for a reload race so I didn't get back to it and now I'm too tired to finish it tonight.